### PR TITLE
Develop > Smart Contracts > Native EVM contracts

### PR DESCRIPTION
**Pull request by nhussein11**
This PR aims to add the "Native EVM contracts" on the smart contracts section, to explain how to use smart contracts on asset hub

### CODE REVIEW

File: develop/smart-contracts/evm/native-evm-contracts.md [hunk 1]
```diff
@@ -1,4 +1,98 @@
...
+
+```mermaid
+graph TD
+    A[pallet_revive] -->|Executes contracts| B[PolkaVM]
+    A -->|Compiles Solidity| C[Revive]
+    C -->|Compiles through| D[Revive Remix]
+    B -->|Handles transactions| E[Ethereum-style Transactions Handling]
+    E --> F[Transaction Conversion & Dispatch]
+    B -->|Executes Solidity code| G[Solidity Execution]
+```
```
Comment (0xLucca) [inline develop/smart-contracts/evm/native-evm-contracts.md:39]: I would remove this diagram, it is kinda confussing

File: develop/smart-contracts/evm/native-evm-contracts.md [hunk 2]
```diff
@@ -1,4 +1,98 @@
...
+
+- [**`pallet_revive`**](https://paritytech.github.io/polkadot-sdk/master/pallet_revive/index.html){target=\_blank} - executes smart contracts by adding extrinsics, runtime APIs, and logic for processing Ethereum-style transactions. Instead of being submitted directly to the blockchain, these transactions are sent through a proxy server that emulates the Ethereum JSON RPC interface. The proxy converts Ethereum transactions into a special dispatchable, leaving the payload intact. The pallet's logic decodes and transforms these transactions into a format compatible with the blockchain, simplifying tooling integration. Using a proxy avoids modifying the node binary, ensuring compatibility with alternative clients without requiring additional implementation
+- [**PolkaVM**](https://github.com/paritytech/polkavm){target=\_blank} - a custom virtual machine replacing EVM, featuring a RISC-V-based register architecture for efficient hardware translation and 64-bit word size for faster arithmetic. Includes an interpreter for lightweight tasks and plans for a JIT compiler, allowing Solidity to interact with high-performance languages
```
Comment (0xLucca) [inline develop/smart-contracts/evm/native-evm-contracts.md:27]:
```suggestion
- [**PolkaVM**](https://github.com/paritytech/polkavm){target=\_blank} - a custom virtual machine, featuring a RISC-V-based register architecture for efficient hardware translation and 64-bit word size for faster arithmetic. Includes an interpreter for lightweight tasks and plans for a JIT compiler, allowing Solidity to interact with high-performance languages
```

File: develop/smart-contracts/evm/native-evm-contracts.md [hunk 3]
```diff
@@ -1,4 +1,98 @@
...
+
+PolkaVM introduces a multi-dimensional gas model to address these differences, incorporating resources like computation time, proof size, and storage deposits. This provides a more accurate representation of contract execution costs and reduces overcharging for memory allocations. PolkaVM also avoids Ethereum’s approach of using a single resource for gas, making it possible to meter resources at the transaction level and for cross-contract calls. For compatibility, PolkaVM’s RPC interface closely mirrors Ethereum’s, with minor adjustments needed for some operations. Additionally, PolkaVM hides the existential deposit requirement to ensure smooth operation across networks, ensuring users aren’t surprised by the underlying balance limitations.
+
+## Get Started
+
+### Connect to Asset Hub
+
+Install the MetaMask browser extension and add a custom network using the settings provided below:
```
Comment (0xLucca) [inline develop/smart-contracts/evm/native-evm-contracts.md:51]: Maybe you can say something like: Install any EVM compatible wallet like metamask

**Review by dawnkelly09 — COMMENTED** [review]
Did a grammar pass and some formatting and polishing on this.

**Comment by nhussein11** [comment]
Re-requesting a review to obtain approvals

File: develop/smart-contracts/evm/native-evm-contracts.md [hunk 4]
```diff
@@ -1,4 +1,110 @@
...
+PolkaVM is a cutting-edge virtual machine tailored to optimize smart contract execution on Polkadot. Unlike traditional Ethereum Virtual Machines (EVM), PolkaVM is built with a RISC-V-based register architecture and a 64-bit word size, enabling:
+
+- Faster arithmetic operations and efficient hardware translation
+- Seamless integration of high-performance languages like C and Rust for advanced optimization
+- Improved scalability for modern blockchain applications
+
+### Compared to Traditional EVMs
+
+- **Architecture** - PolkaVM's register-based design offers significant performance improvements over Ethereum's stack-based EVM. It allows for faster compilation times and aligns better with modern hardware, reducing bottlenecks in contract execution
+- **Gas Modeling** - PolkaVM employs a multi-dimensional gas model, metering resources like computation time, storage, and proof sizes. This ensures more accurate cost assessments for contract execution, reducing overcharging for memory allocation and enabling efficient cross-contract calls
```
Comment (eshaben) [inline develop/smart-contracts/evm/native-evm-contracts.md:45]:
```suggestion
- **Gas modeling** - PolkaVM employs a multi-dimensional gas model, metering resources like computation time, storage, and proof sizes. This ensures more accurate cost assessments for contract execution, reducing overcharging for memory allocation and enabling efficient cross-contract calls
```

**Review by eshaben — COMMENTED** [review]
Looks good!

**Review by eshaben — APPROVED** [review]
LGTM!

**Review by 0xLucca — APPROVED** [review]
LGTM!

**Pull request merged**